l1-contracts/contracts/bridge/L1ERC20Bridge.sol

```diff
@@ -2,133 +2,79 @@
-        internal depositAmount;
+        public depositAmount;
 
-    /// @dev The address of deployed L2 bridge counterpart
+    /// @dev The address that is used as a L2 bridge counterpart in zkSync Era.
     address public l2Bridge;
 
-    /// @dev The address that acts as a beacon for L2 tokens
+    /// @dev The address that is used as a beacon for L2 tokens in zkSync Era.
     address public l2TokenBeacon;
 
-    /// @dev The bytecode hash of the L2 token contract
+    /// @dev Stores the hash of the L2 token proxy contract's bytecode on zkSync Era.
     bytes32 public l2TokenProxyBytecodeHash;
 
+    /// @dev Deprecated storage variable related to withdrawal limitations.
     mapping(address => uint256) private __DEPRECATED_lastWithdrawalLimitReset;
 
-    /// @dev A mapping L1 token address => the accumulated withdrawn amount during the withdrawal limit window
+    /// @dev Deprecated storage variable related to withdrawal limitations.
     mapping(address => uint256) private __DEPRECATED_withdrawnAmountInWindow;
 
-    /// @dev The accumulated deposited amount per user.
-    /// @dev A mapping L1 token address => user address => the total deposited amount by the user
+    /// @dev Deprecated storage variable related to deposit limitations.
     mapping(address => mapping(address => uint256)) private __DEPRECATED_totalDepositedAmountPerUser;
 
     /// @dev Contract is expected to be used as proxy implementation.
     /// @dev Initialize the implementation to prevent Parity hack.
-    constructor(IZkSync _zkSync) reentrancyGuardInitializer {
-        zkSync = _zkSync;
+    constructor(IL1SharedBridge _sharedBridge) reentrancyGuardInitializer {
+        sharedBridge = _sharedBridge;
     }
 
-    /// @dev Initializes a contract bridge for later use. Expected to be used in the proxy
-    /// @dev During initialization deploys L2 bridge counterpart as well as provides some factory deps for it
-    /// @param _factoryDeps A list of raw bytecodes that are needed for deployment of the L2 bridge
-    /// @notice _factoryDeps[0] == a raw bytecode of L2 bridge implementation
-    /// @notice _factoryDeps[1] == a raw bytecode of proxy that is used as L2 bridge
-    /// @notice _factoryDeps[2] == a raw bytecode of token proxy
-    /// @param _l2TokenBeacon Pre-calculated address of the L2 token upgradeable beacon
-    /// @notice At the time of the function call, it is not yet deployed in L2, but knowledge of its address
-    /// @notice is necessary for determining L2 token address by L1 address, see `l2TokenAddress(address)` function
-    /// @param _governor Address which can change L2 token implementation and upgrade the bridge
-    /// @param _deployBridgeImplementationFee How much of the sent value should be allocated to deploying the L2 bridge
-    /// implementation
-    /// @param _deployBridgeProxyFee How much of the sent value should be allocated to deploying the L2 bridge proxy
-    function initialize(
```

Comment:
why this method has been deleted while the fields `l2TokenProxyBytecodeHash` , `l2TokenBeacon`  are still there but cannot be initialized anywhere?

Reply:
Because we moved most of the functionality to the L1SharedBridge, and this will be kept as a legacy bridge.  This keeps the code cleaner. On the shared bridge we will not have these interfaces, and on this bridge these variables are already set.

Reply:
I see, but in that case, should we also create expose a `l2TokenAddress()` method on the L1SharedBridge? 
Currently there is no good way to find out what is l2 Token Address unless we used this method.

This methods depends on `l2TokenProxyBytecodeHash` and `l2TokenBeacon` that was set initially in L1ERC20Bridge.

For bridge developper that what to develop their own adapters, should we rely on an implementation reference?

Reply:
To find l2TokenAddress off-chain you can call the layer2. There should be no need to know the l2TokenAddress on L1, you can identify the token based on l1TokenAddress, and you can messages to the token via the bridge. 

The L1ERC20Bridge will from now on be a legacy bridge, the L1SharedBridge should be the one that external contracts use. We also recommend doing additional bridges less from now on, it makes sense to have a central SharedBridge, with customisable tokens